code/datums/components/nanites.dm

```diff
@@ -0,0 +1,307 @@
+/datum/component/nanites
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+
+	var/mob/living/host_mob
+	var/nanite_volume = 100		//amount of nanites in the system, used as fuel for nanite programs
+	var/max_nanites = 500		//maximum amount of nanites in the system
+	var/regen_rate = 0.5		//nanites generated per second
+	var/safety_threshold = 50	//how low nanites will get before they stop processing/triggering
+	var/cloud_id = 0 			//0 if not connected to the cloud, 1-100 to set a determined cloud backup to draw from
+	var/next_sync = 0
+	var/list/datum/nanite_program/programs = list()
+	var/max_programs = NANITE_PROGRAM_LIMIT
+
+	var/stealth = FALSE //if TRUE, does not appear on HUDs and health scans, and does not display the program list on nanite scans
+
+/datum/component/nanites/Initialize(amount = 100, cloud = 0)
+	nanite_volume = amount
+	cloud_id = cloud
+
+	RegisterSignal(parent, COMSIG_HAS_NANITES, .proc/confirm_nanites)

```

Comment:
Move all your signal registrations into the `RegisterWithParent()` and unregister them in `UnregisterFromParent()`. Because of how signal registration got changed it isnt easy to move signal registry over to new parents in transfer so you need to handle that yourself in these procs.